cms/cms_menus.py

```diff
@@ -198,6 +198,46 @@ class CMSMenu(Menu):
     based on a site's :class:`cms.models.pagemodel.Page` objects.
     """
 
+    @staticmethod
+    def get_pagecontent_queryset(toolbar):
+        """
+        Returns a queryset of page content objects based on the given toolbar: Includes
+        latest content in edit and preview mode.
+
+        Parameters:
+        - toolbar (Toolbar): The toolbar object passed as an argument.
+
+        Returns:
+        - A queryset of page content objects.
+        """
+        if toolbar.edit_mode_active or toolbar.preview_mode_active:
+            # Get all page contents visible in the admin
+            return PageContent.admin_manager.current_content()
+        else:
+            # Only get public page contents
+            return PageContent.objects.filter()
+
+    @staticmethod
+    def preview_endpoint(toolbar):
+        """
+        This method is used to get a callable based on the state of the toolbar which
+        is used to create the preview endpoint URL for each page content in the menu.
+        The callable must accept a page content object as a single parameter.
+
+        If the preview mode or edit mode is active in the toolbar, the callable will be set to the
+        get_object_preview_url function. Otherwise, it will be set to None leading to the fallback
+        of the get_absolute_url method of the page content.
+
+        :param toolbar: The toolbar object that contains the state information.
```

Comment:
We've got inconsistent docstring formats here. 🤔

Reply:
Thanks, @marksweb ! I update the docstrings.